crates/compiler/fmt/tests/test_fmt.rs

```diff
@@ -5705,6 +5705,39 @@ mod test_fmt {
         ));
     }
 
+    #[test]
+    fn clauses_with_multiple_abilities() {
+        expr_formats_same(indoc!(
+            r#"
+            f : {} -> a | a has Eq & Hash & Decode
```

Comment:
Syntactically, I think this would still look good with some of the other ideas we've talked about, e.g. in the most verbose case among the options we've discussed:

```elm
f : {} -> a where a implements Eq & Hash & Decode
```